so/include/so/G3SuperTimestream.h

```diff
@@ -0,0 +1,123 @@
+public:
+    G3SuperTimestream();
+    ~G3SuperTimestream();
+
+    G3SuperTimestream(const G3VectorString &names_, const G3VectorTime &times_);
+    G3SuperTimestream(const G3VectorString &names_, const G3VectorTime &times_,
+              const bp::object &data);
+    G3SuperTimestream(const G3VectorString &names_, const G3VectorTime &times_,
+              const bp::object &data_, const std::vector<double> &quanta_);
+
+    // This object contains pointers to memory that it
+    // allocated... and they're freed on destruction.  The
+    // responsible thing to do in such circumstances is to delete
+    // or modify the move and copy constructors.  But if we do
+    // that, boost python complains about something.  In G3 you
+    // avoid move/copy by wrapping all instances in a G3xxxPtr,
+    // and essentially pass the object around by reference, so
+    // that's what we'll do.  But beware, there's nothing
+    // preventing you from returning G3SuperTimestream directly
+    // from a function (except the inevitable segfault).
+
+    string Description() const;
+    string Summary() const;
+
+    bool Extract(bp::object dest, bp::object dest_indices, bp::object src_indices,
+             int start, int stop);
+    bool Encode();
+    bool Decode();
+    void Calibrate(vector<double> rescale);
+    int Options(int enable=-1,
+            int flac_level=-1, int bz2_workFactor=-1,
+                    int data_algo=-1, int times_algo=-1);
+
+    // Interface for C++...
+    bool SetDataFromBuffer(void* buf, int ndim, int shape[], int typenum,
+                   std::pair<int,int> sample_range);
+
+
+    template <class A> void load(A &ar, unsigned v);
+    template <class A> void save(A &ar, unsigned v) const;
+
+    struct array_desc {
+        npy_intp type_num;
+        npy_intp ndim;
+        npy_intp shape[32];
+        npy_intp nbytes;
+    };
+
+    // Container for the compressed data.
+    struct array_blob {
+        int size;
+        char *buf;
+        int count;
+        vector<int> offsets;
+    };
+
+    enum algos {
+        ALGO_NONE = 0,
+        ALGO_DO_FLAC = (1 << 0),
+        ALGO_DO_BZ = (1 << 1),
+        ALGO_DO_CONST = (1 << 2)
+    };
+
+    struct options_type {
+        int8_t times_algo;
+        int8_t data_algo;
+        int8_t flac_level;
+        int8_t bz2_workFactor;
+    } options;
+
+    G3VectorTime times;
+    G3VectorString names;
+
+    bool float_mode;
+    bool dataful;
+    vector<double> quanta;
+    struct array_desc desc;
+
+    PyArrayObject *array;
+    struct array_blob *ablob;
+};
+
+// This specialization tells cereal to use G3SuperTimestream::load/save
+// and not the base class' load/save.
+namespace cereal {
+    template <class A> struct specialize<
+        A, G3SuperTimestream, cereal::specialization::member_load_save> {};
+}
+
+G3_POINTERS(G3SuperTimestream);
+G3_SERIALIZABLE(G3SuperTimestream, 0);
+
+class g3supertimestream_exception : std::exception
```

Comment:
use the G3Logging stuff instead of creating a custom exception -- `SET_LOGGER("G3SuperTimestream")` in the class definition, and use `log_fatal()` instead of `throw g3supertimestream_exception`